tee-worker/omni-executor/intent/executors/cross-chain/src/lib.rs

```diff
@@ -268,16 +255,28 @@ impl<BinanceClient: BinanceApi, SolanaClient: SolanaClientTrait> IntentExecutor
 						&access_token,
 						amount,
 						pumpx_config,
+						from_address,
 					)
 					.await?;
 
-				// self.account_asset_lock.release(
-				// 	account_id.clone(),
-				// 	swap_order.from_asset.clone(),
-				// 	AmountType::from_str_radix(&from_amount_string, 10).map_err(|_| {
-				// 		log::error!("Failed to parse from_amount_string");
-				// 	})?,
-				// )?;
+				// depost here and unlock assets
+				if let Some(details) = instant_flow_details {
```

Comment:
Hmm this is still synchronous - can we push the task into a queue which is processed in (another) dedicated thread and return as early as possible?

Does it make sense?
Not necessarily in this PR

Reply:
It should make intent processing faster - we can add such improvement in separate PR.

Reply:
I've managed to do that in this PR - should be resolved now.